letter-shop.md

```diff
@@ -0,0 +1,232 @@
+# Letter Shop Tutorial
+
+## What you need before you start:
+
+* a laptop with an internet connection and NodeJS installed
+* basic knowledge of the command line terminal
+* basic knowledge of JavaScript, including Promises and Buffers
+* (optional) the concept of one-way hash functions
+
+## What you'll learn:
+
+* the concept Hash Time Lock Agreements (HTLAs)
+* the Ledger Plugin Interface (LPI)
+* how to build a paid web app using Interledger!
+
+## Step 1:
+
+Save the following JavaScript as `shop.js`:
+
+```js
+const http = require('http')
+const crypto = require('crypto')
+const Plugin = require('ilp-plugin-xrp-escrow')
+function base64 (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+
+let fulfillments = {}
+let letters = {}
+
+const plugin = new Plugin({
+  secret: 'ssGjGT4sz4rp2xahcDj87P71rTYXo',
+  account: 'rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+plugin.connect().then(function () {
+  plugin.on('incoming_prepare', function (transfer) {
+    plugin.fulfillCondition(transfer.id, fulfillments[transfer.executionCondition]).catch(function () {})
+  })
+
+  http.createServer(function (req, res) {
+    if (letters[req.url.substring(1)]) {
+      res.end('Your letter: ' + letters[req.url.substring(1)])
+    } else {
+      const secret = crypto.randomBytes(32)
+      const fulfillment = base64(secret)
+      const condition = base64(crypto.createHash('sha256').update(secret).digest())
+      const letter = ('ABCDEFGHIJKLMNOPQRSTUVWXYZ').split('')[(Math.floor(Math.random() * 26))]
+      fulfillments[condition] = fulfillment
+      letters[fulfillment] = letter
+      console.log('Generated letter for visitor on ', req.url, { secret, fulfillment, condition, letter })
+      res.end('Please send an Interledger payment to ' + plugin.getAccount() + ' with condition ' + condition)
+    }
+  }).listen(8000)
+})
+```
+
+Set up a Letter Shop website on http://localhost:8000, by running:
+
+```sh
+npm install michielbdejong/ilp-plugin-xrp-escrow#3fadeb4
+node shop.js
+```
+In the code, you see that an 'ilp-plugin-xrp-escrow' Plugin is being configured with secret, account, server, and prefix.
```

Comment:
These are very dense paragraphs. It would be good to break them up into much shorter sections and maybe put some headings above them to help when the reader is skimming